metapackages/auto-instrumentations-node/src/utils.ts

```diff
@@ -253,3 +262,56 @@ export function getResourceDetectorsFromEnv(): Array<Detector | DetectorSync> {
     return resourceDetector || [];
   });
 }
+
+type PropagatorFactoryFunction = () => TextMapPropagator;
+
+const propagatorMap = new Map<string, PropagatorFactoryFunction>([
+  ['tracecontext', () => new W3CTraceContextPropagator()],
+  ['baggage', () => new W3CTraceContextPropagator()],
+  [
+    'b3',
+    () => new B3Propagator({ injectEncoding: B3InjectEncoding.SINGLE_HEADER }),
+  ],
+  [
+    'b3multi',
+    () => new B3Propagator({ injectEncoding: B3InjectEncoding.MULTI_HEADER }),
+  ],
+  ['jaeger', () => new JaegerPropagator()],
+  ['xray', () => new AWSXRayPropagator()],
+  ['ottrace', () => new OTTracePropagator()],
+]);
+
+/**
+ * Get a propagator based on the OTEL_PROPAGATORS env var.
+ */
+export function getPropagator(): TextMapPropagator {
+  if (process.env.OTEL_PROPAGATORS == null) {
+    return new CompositePropagator({
+      propagators: [
+        new W3CTraceContextPropagator(),
+        new W3CBaggagePropagator(),
+      ],
+    });
+  }
+
+  const propagatorsFromEnv = process.env.OTEL_PROPAGATORS?.split(',').map(
```

Comment:
FWIW, `OTEL_PROPAGATORS` parsing used by sdk-node trims each entry to allow `foo, bar` with spaces: https://github.com/open-telemetry/opentelemetry-js/blob/4e778b1257c8e8b7fbeb33e8eeba22b78ac33dd6/packages/opentelemetry-core/src/utils/environment.ts#L303

It also dedupes values: https://github.com/open-telemetry/opentelemetry-js/blob/4e778b1257c8e8b7fbeb33e8eeba22b78ac33dd6/packages/opentelemetry-sdk-trace-base/src/BasicTracerProvider.ts#L240-L243

Reply:
good point, aligned the behavior in https://github.com/open-telemetry/opentelemetry-js-contrib/pull/2232/commits/c67eda1cd17374a7c7e56d4ce97c2c96c3024259 :slightly_smiling_face: